core/unit_test/TestCrs.hpp

```diff
@@ -63,11 +63,35 @@ struct CountFillFunctor {
   }
 };
 
+template< class CrsType, class ExecSpace, class scalarType >
+struct RunUpdateCrsTest {
+
+  CrsType graph {};
+  RunUpdateCrsTest( CrsType g_in ) : graph(g_in)
+  {
+  }
+
+  void run_test() {
+     parallel_for ("TestCrs1", Kokkos::RangePolicy<ExecSpace>(0,graph.numRows()),*this);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const scalarType row) const {
+     auto row_map = graph.row_map;
+     auto entries = graph.entries;
+     auto j_start = (row >= static_cast<decltype(row)>(row_map.extent(0))) ? 0 : row_map(row);
+     auto j_end = (row+1 >= static_cast<decltype(row)>(row_map.extent(0))) ? 0 : row_map(row+1)-j_start;
+     for (scalarType j = 0; j < j_end; ++j) {
+        entries(j_start+j) = (j+1)*(j+1);
+     }
+  }
+};
+
```

Comment:
We need 4 versions of this test: 
 * what you got now
 * One where as the first thing it copy constructs the graph `CrsType local_graph(graph)` and then uses that to get row_map etc.
 * One where as the first thing it default constructs the graph and then assigns it `CrsType local_graph; local_graph = graph` and then uses that to get row_map etc.
 * One where as the first thing it constructs a local graph from the row_map and entries `CrsType local_graph(graph.row_map,graph.entries)` and then uses that to get row_map etc.

Reply:
You can use Tagged operators to do it in the same class. You may even want to take your current operator and make it a function which takes a CrsGraph and a row.